apps/common/js/components/instructions-overlay.jsx

```diff
@@ -3,13 +3,21 @@ import LeapHandsView from './leap-hands-view.jsx';
 import '../../css/instructions-overlay.less';
 
 export default class InstructionsOverlay extends React.Component {
-
+  componentWillMount() {
+    this.setState({ handsOpacity: this.props.handsOpacity });
+  }
+  componentWillReceiveProps(newProps) {
+    if (newProps.handsOpacity != this.props.handsOpacity) {
+      this.setState({ handsOpacity: newProps.handsOpacity });
+    }
+  }
```

Comment:
I find this relation between props and state a bit confusing. I would either keep it in state or props. And actually state.handsOpacity seems to be unused anyway.

I'm assuming that you left that to possibly fade out instructions in the future. I think it might be achieved easier using simple CSS transitions (and faster, no react machinery involved). That's how it is implemented in MW examples.